src/Sema.zig

```diff
@@ -16515,10 +16515,9 @@ fn zirTypeInfo(sema: *Sema, block: *Block, inst: Zir.Inst.Index) CompileError!Ai
                     .val = if (is_generic) .none else param_ty,
                 } });
 
-                const is_noalias = blk: {
-                    const index = std.math.cast(u5, i) orelse break :blk false;
```

Comment:
`orelse break :blk false` is very sus here. I think this may be a bug.

Reply:
No, this logic is correct - our compiler implementation only allows the first 32 parameters to be marked `noalias`. (But the new approach is a bit cleaner anyway.)

Reply:
The new approach simply crashes. I don't think it's cleaner.

Reply:
Oh whoops, I didn't see that it was `@intCast` to `u5`. You still need to handle that case manually. Best would probably be:
```
const is_noalias = if (std.math.cast(u5, i)) |idx| func_ty_info.paramIsNoalias(idx) else false;
```
...and same for `is_comptime`.